scripts/storage-emulator-integration/conformance/gcs-js-sdk.test.ts

```diff
@@ -0,0 +1,961 @@
+import { Bucket, CopyOptions } from "@google-cloud/storage";
+import { expect } from "chai";
+import * as admin from "firebase-admin";
+import * as fs from "fs";
+import { EmulatorEndToEndTest } from "../../integration-helpers/framework";
+import { TEST_ENV } from "./env";
+import {
+  createRandomFile,
+  EMULATORS_SHUTDOWN_DELAY_MS,
+  resetStorageEmulator,
+  SMALL_FILE_SIZE,
+  TEST_SETUP_TIMEOUT,
+  getTmpDir,
+} from "../utils";
+
+// TODO(b/241151246): Fix conformance tests.
+describe("GCS Javascript SDK conformance tests", () => {
+  // Temp directory to store generated files.
+  const tmpDir = getTmpDir();
+  const smallFilePath: string = createRandomFile("small_file", SMALL_FILE_SIZE, tmpDir);
+
+  const storageBucket = TEST_ENV.appConfig.storageBucket;
+  const storageHost = TEST_ENV.storageHost;
+
+  let test: EmulatorEndToEndTest;
+  let testBucket: Bucket;
+
+  async function resetState(): Promise<void> {
+    if (TEST_ENV.useProductionServers) {
+      await testBucket.deleteFiles();
+    } else {
+      await resetStorageEmulator(TEST_ENV.storageEmulatorHost);
+    }
+  }
+  before(async function (this) {
+    this.timeout(TEST_SETUP_TIMEOUT);
+    TEST_ENV.applyEnvVars();
+    if (!TEST_ENV.useProductionServers) {
+      test = new EmulatorEndToEndTest(TEST_ENV.fakeProjectId, __dirname, TEST_ENV.emulatorConfig);
+      await test.startEmulators(["--only", "storage"]);
+    }
+
+    // Init GCS admin SDK.
+    const credential = TEST_ENV.prodServiceAccountKeyJson
+      ? admin.credential.cert(TEST_ENV.prodServiceAccountKeyJson)
+      : admin.credential.applicationDefault();
+    admin.initializeApp({ credential });
+    testBucket = admin.storage().bucket(storageBucket);
+  });
+
+  beforeEach(async () => {
+    await resetState();
+  });
+
+  after(async function (this) {
+    this.timeout(EMULATORS_SHUTDOWN_DELAY_MS);
+    admin.app().delete();
+    fs.rmSync(tmpDir, { recursive: true, force: true });
+
+    TEST_ENV.removeEnvVars();
+    if (!TEST_ENV.useProductionServers) {
+      await test.stopEmulators();
+    }
+  });
+
+  describe(".bucket()", () => {
+    describe("#upload()", () => {
+      it("should handle non-resumable uploads", async () => {
```

Comment:
Would be good to add a js sdk resumable upload test too since thats a pretty standard flow. Could you add a TODO for now? We can add in a later pr

Reply:
added in a test